pkg/controllers/resources/volumesnapshots/volumesnapshotcontents/syncer.go

```diff
@@ -0,0 +1,166 @@
+	log.Infof("create VolumeSnapshotContent %s, because it does not exist in the virtual cluster", vVSC.Name)
+	return ctrl.Result{}, s.virtualClient.Create(ctx, vVSC)
+}
+
+func (s *syncer) Forward(ctx context.Context, vObj client.Object, log loghelper.Logger) (ctrl.Result, error) {
+	// We are not doing any syncing Forward for the VolumeSnapshotContentes
+	// if this method is called it means that VolumeSnapshotContent was deleted in host,
+	// or a new VolumeSnapshotContent was created in vcluster, and it should be deleted.
+
+	//remove finalizers if present on vObj
+	if len(vObj.GetFinalizers()) > 0 {
+		vObj.SetFinalizers([]string{})
+		err := s.virtualClient.Update(ctx, vObj)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	log.Infof("delete VolumeSnapshotContent %s, because it does not exist in the host cluster", vObj.GetName())
+	zero := int64(0)
+	return ctrl.Result{}, s.virtualClient.Delete(ctx, vObj, &client.DeleteOptions{
+		GracePeriodSeconds: &zero,
+	})
+}
+
+func (s *syncer) Update(ctx context.Context, pObj client.Object, vObj client.Object, log loghelper.Logger) (ctrl.Result, error) {
+	pVSC := pObj.(*volumesnapshotv1.VolumeSnapshotContent)
+	vVSC := vObj.(*volumesnapshotv1.VolumeSnapshotContent)
+
+	//check if objects are getting deleted
+	if pObj.GetDeletionTimestamp() != nil {
+		if vObj.GetDeletionTimestamp() == nil {
+			log.Infof("delete virtual VolumeSnapshotContent %s, because physical VolumeSnapshotContent is being deleted", vObj.GetName())
+			err := s.virtualClient.Delete(ctx, vObj, &client.DeleteOptions{
+				GracePeriodSeconds: pObj.GetDeletionGracePeriodSeconds(),
+			})
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	// check if the VolumeSnapshotContent should get synced
+	sync, vVS, err := s.shouldSync(ctx, pVSC)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if !sync {
+		// ignore this VolumeSnapshotContent resource, because there is no virtual VolumeSnapshot bound to it"
+		return ctrl.Result{}, nil
+	}
+
+	updatedObj := s.translateUpdateBackwards(pVSC, vVSC, vVS)
+	if updatedObj != nil {
+		log.Infof("update virtual VolumeSnapshotContent %s, because spec or metadata(annotations or labels) have changed", pVSC.Name)
+		err = s.virtualClient.Update(ctx, updatedObj)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	// check status
+	if !equality.Semantic.DeepEqual(vVSC.Status, pVSC.Status) {
+		vVSC.Status = pVSC.Status.DeepCopy()
+		log.Infof("update virtual VolumeSnapshotContent %s, because status has changed", pVSC.Name)
+		err = s.virtualClient.Status().Update(ctx, vVSC)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (s *syncer) shouldSync(ctx context.Context, pObj *volumesnapshotv1.VolumeSnapshotContent) (bool, *volumesnapshotv1.VolumeSnapshot, error) {
+	if pObj.Spec.VolumeSnapshotRef.Namespace != s.targetNamespace {
+		return false, nil, nil
```

Comment:
I guess we should also allow snapshot contents that were created in the virtual cluster

Reply:
Added in the second iteration that I just pushed. :)